### PR TITLE
Lerna

### PR DESCRIPTION
**Pull request by ikesau**
Adds lerna to the top of the repo to manage three new sub packages:
- `@ourworldindata/utils`
- `@ourworldindata/core-table`
- `@ourworldindata/grapher`

There's also an empty `@ourworldindata/types` package which is unused at the moment but I intend to add to later.

Lerna works best when every project is in a workspace it's watching, but for now we still have the site, client, server, etc outside of Lerna's scope. 

This means that our top level `package.json` still has the dependencies for all these projects and they all have to be installed with `yarn install` before running `npx lerna bootstrap`

To test, all you should need to do is:
1. `rm -rf node_modules`
2. `make up.full`

The only business logic change to the project is that we now no longer bake Grapher with its URL dependencies hard-coded, they must be [passed via config during instantiation instead](https://github.com/owid/owid-grapher/commit/df315c62da150d90a084efe39da6ee5e695f327e). 

`utils` and `core-table` both build with `tsc` to output a declaration and js files in their `dist/` directories. `grapher` uses [tsup](https://tsup.egoist.dev) which is an esbuild wrapper in order to also bundle the CSS.

esbuild runs extremely quickly if you opt to skip type checking. ~I still need to implement a `watch` script for local grapher development that will try to leverage this.~

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "$schema": "node_modules/lerna/schemas/lerna-schema.json",
+    "useWorkspaces": true,
```
Comment (marcelgerber) [inline lerna.json:3]:
```suggestion
    "npmClient": "yarn",
    "useWorkspaces": true,
```

This [seems to](https://www.honeybadger.io/blog/monorepo-yarn-workspace-lerna/) make lerna use yarn, have you tried it?

File: packages/@ourworldindata/core-table/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "keywords": [],
+  "author": "",
+  "dependencies": {
+    "@ourworldindata/utils": "*",
+    "@types/d3": "^6",
+    "d3": "^6.1.1",
+    "dayjs": "^1.11.5",
+    "eslint": "^8.23.1",
+    "fast-cartesian": "^5.1.0",
+    "prettier": "^2.7.0",
```
Comment (marcelgerber) [inline packages/@ourworldindata/core-table/package.json:23]: nit: do we really need `prettier` in every module? don't think so.

File: packages/@ourworldindata/core-table/package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "typescript": "~4.8.2"
```
Comment (marcelgerber) [inline packages/@ourworldindata/core-table/package.json:24]: also a nit, and we don't need to address this now: but we can make our split between `dependencies` and `devDependencies` more sensible now (and should do this before we publish to npm), and typescript is surely a `devDependency`.

File: packages/@ourworldindata/core-table/package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+},
+  "license": "ISC"
```
Comment (marcelgerber) [inline packages/@ourworldindata/core-table/package.json:26]: We didn't use that license beforehand, correct?

**Review by marcelgerber — COMMENTED** [review]
This is an absolutely heroic effort! 🙌🏻 

I'm mostly concerned about the use of two different package managers, with slightly differing resolution algorithms and different lockfiles. We should find a way to use only one or the other, and it seems like there is one.

File: packages/@ourworldindata/core-table/package.json (hunk 4 above)
Comment (ikesau) [reply packages/@ourworldindata/core-table/package.json]: Updated all license fields in our package.jsons (and bumped the year in LICENSE.md)

File: Makefile [hunk 5]
```diff
@@ -90,6 +98,8 @@ up.full: require create-if-missing.env.full wordpress/.env tmp-downloads/owid_ch
 
 	@echo '==> Building grapher'
 	yarn install
+	npx lerna bootstrap
+	npx lerna run build
```
Comment (mlbrgl) [inline Makefile:102]: the first time `lerna run build` runs, it fails. As I understood it, all builds run in parallel, which means that e.g. utils is not compiled by the time grapher or core-table needs it.

File: package.json [hunk 6]
```diff
@@ -11,6 +11,7 @@
         "buildLocalBake": "node ./itsJustJavascript/baker/buildLocalBake.js",
         "buildStorybook": "build-storybook -c ./itsJustJavascript/.storybook -o .storybook/build",
         "buildTsc": "tsc -b -verbose",
+        "buildLerna": "lerna run build",
         "buildWebpack": "rm -rf itsJustJavascript/webpack && ENV=production webpack --mode production --config ./itsJustJavascript/webpack.config.js --progress",
         "buildWordpressPlugin": "cd ./wordpress/web/app/plugins/owid && yarn && yarn build",
         "cleanTsc": "rm -rf itsJustJavaScript && tsc -b -clean",
```
[the remark this answers is not in the document]
Comment (mlbrgl) [reply package.json:17]: as well as nx cache

**Review by mlbrgl — COMMENTED** [review]
Thanks for taking this on! It looks like quite an effort to pull this together. 

I have yet to fully test the debugging setup in VSCode but it looks like creating a task and making it part of the "Launch site dev" debug flow should be fine. I might do that post merge.

File: Makefile (hunk 5 above)
Comment (ikesau) [reply Makefile:102]: This shouldn't be the case. Nx understands the build's dependencies and sequences them appropriately.

I tried to reproduce this failure:
1. `git checkout master` 
2. `rm -rf packages node_modules itsJustJavascript`
3. `git checkout lerna`
4. `make up.full`

And it spun up without complaint, building utils, then core-table, then grapher.

Could you please try to reproduce the failure? 🙏

Comment (mlbrgl) [reply Makefile]: @ikesau sending you a screen grab

**Review by marcelgerber — APPROVED** [review]

**Pull request merged**